lib/state/ui/reducer.ts

```diff
@@ -52,24 +51,48 @@ const simperiumConnected: A.Reducer<boolean> = (state = false, action) =>
     ? action.simperiumConnected
     : state;
 
-const visiblePanes: A.Reducer<string[]> = (
+const visiblePanes: A.Reducer<Set<string>> = (
   state = defaultVisiblePanes,
   action
 ) => {
-  if ('TAG_DRAWER_TOGGLE' === action.type) {
-    return action.show
-      ? union(state, ['tagDrawer'])
-      : difference(state, ['tagDrawer']);
+  switch (action.type) {
+    case 'CLOSE_NOTE': {
+      return new Set(state).add('noteList');
+    }
+    case 'App.selectNote': {
+      const newSet = new Set(state);
+      newSet.delete('noteList');
+      return newSet;
+    }
+    case 'NOTE_LIST_TOGGLE':
+      if (action.show) {
+        return new Set(state).add('noteList');
+      } else {
+        const newSet = new Set(state);
+        newSet.delete('noteList');
+        return newSet;
+      }
+    case 'TAG_DRAWER_TOGGLE': {
+      if (action.show) {
+        return new Set(state).add('tagDrawer');
+      } else {
+        const newSet = new Set(state);
+        newSet.delete('tagDrawer');
+        return newSet;
```

Comment:
as noted in Slack, we're going to remove `visiblePanes` right?

Reply:
yes, I hadn't updated yet.